### PR TITLE
L1TRawToDigi Fixes

### PR DESCRIPTION
**Pull request by Martin-Grunewald**
L1TRawToDigi Fixes
Includes a fixed version of @mulhearn 's https://github.com/cms-sw/cmssw/pull/13549
and thus replaces that PR; and a fix by @blwiner on the number of algos being unpacked.

### CODE REVIEW

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by mulhearn** [comment]
+1

File: EventFilter/L1TRawToDigi/src/implementations_stage2/GlobalAlgBlkUnpacker.cc [hunk 1]
```diff
@@ -77,8 +77,8 @@ namespace stage2 {
          if( (block.header().getID()!=initialBlkID+4 && block.header().getID()!=prescaledBlkID+4 && block.header().getID()!=finalBlkID+4 ) || wd<4) {
            for(unsigned int bt=0; bt<32; bt++) {
 	     int val = ((raw_data >> bt) & 0x1);
-	     int algBit = bt+wd*32+algOffset;
-             if(val==1 && algBit < 128) { //FIX ME...get dimension from object
+	     unsigned int algBit = bt+wd*32+algOffset;
+             if(val==1 && algBit < alg.maxPhysicsTriggers) { //FIX ME...get dimension from object
```
Comment (fwyzard) [inline EventFilter/L1TRawToDigi/src/implementations_stage2/GlobalAlgBlkUnpacker.cc]: will this affect the behaviour of the HLTL1TSeed modules when we run over data with CMSSW_8_0_1 ?

**Comment by mulhearn** [comment]
Yes, in 801 (until it is patched) the HLTL1TSeed filter will be rejecting events in paths that require triggers above bit 128...

-Mike

**Comment by davidlange6** [comment]
+1

**Comment by davidlange6** [comment]
@Martin-Grunewald @mulhearn - I just notice the data format changes - are they really required? makes a patch impossible

**Comment by Martin-Grunewald** [comment]
What do you mean? Data member variables are unchanged!

**Comment by davidlange6** [comment]
its changing headers in data formats

**Comment by fwyzard** [comment]
I'm afraid this one
```
const static unsigned int maxPhysicsTriggers = 512;
```
is needed.

**Comment by davidlt** [comment]
You have to options:
- Downgrade this to warning by setting CXXFLAGS in BuildFile.xml
- Force compiler to emit it even if it's not used by adding the attribute

**Comment by davidlt** [comment]
Ah, I see `maxPhysicsTriggers` was moved to a different header. Compilation issues are gone.

**Pull request merged**